index.js

```diff
@@ -1 +1,94 @@
-console.log('hello world');
+import express from 'express';
+import { read } from './jsonFileStorage.js';
+
+const app = express();
+
+app.get('/recipe/:index', (req, res) => {
+  // get index number from browser req
+  const index = req.params.index;
+  read('data.json', (err, content) => {
+    if (err) {
+      console.log('Read Error', err);
+    }
+    if (content.recipes[index]) {
+      res.send(content.recipes[index]);
+    } else {
+      res.status(404).send('We cannot find that.');
+    }
+  });
+});
+
+/**
+ * returns based on recipe label
+ */
+app.get('/recipe-label/:label', (req, res) => {
+  // get index number from browser req
+  const { label } = req.params;
+  // replace '-' from LABEL to spaces
+  const labelCorrected = label.replaceAll('-', ' ');
+  read('data.json', (err, content) => {
+    if (err) {
+      console.log('Read Error', err);
+    }
+    const recipes = content.recipes;
+    const filteredRecipes = recipes.filter((recipe) => {
+      return recipe['label'].toLowerCase() === labelCorrected;
+    });
+    let output = '';
+    // add the recipe into OUTPUT
+    filteredRecipes.forEach((recipe) => {
+      for (let [key, value] of Object.entries(recipe)) {
+        output += `${key.toUpperCase()}: `;
+        output += `${value}<br>`;
+      }
+      output += `---<br>`;
+    });
+    res.send(output);
+  });
+});
+
+/**
+ * returns based on yield number
+ */
+app.get('/yield/:number', (req, res) => {
+  // get index number from browser req
+  console.log(`Incoming request..`);
+  // convert param to number
+  const { number } = req.params;
+  read('data.json', (err, content) => {
+    if (err) {
+      console.log('Read Error', err);
+    }
+    // get the array from recipe obj
+    const recipes = content.recipes;
+    // filter the arrays that meet the NUMBER criteria
+    const filteredArray = recipes.filter((recipe) => {
+      return recipe.yield === Number(number);
+    });
```

Comment:
consider naming your variable that gives more meaning to what you are assigning it to.
```suggestion
    const filteredRecipe = recipes.filter((recipe) => {
      return recipe.yield === Number(number);
    });
```